src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -251,6 +259,44 @@ private async Task FillRequestLists(List<Hl7.Fhir.Model.Bundle.EntryComponent> b
             }
         }
 
+        public async Task ResolveIntraBundleReferences(EntryComponent entry, Dictionary<string, (string resourceId, string resourceType)> referenceIdDictionary)
+        {
+            IEnumerable<ResourceReference> references = entry.Resource.GetAllChildren<ResourceReference>();
+
+            foreach (ResourceReference reference in references)
+            {
+                // We've already come across this ID
+                if (referenceIdDictionary.TryGetValue(reference.Reference, out var referenceInformation))
+                {
+                    reference.Reference = $"{referenceInformation.resourceType}/{referenceInformation.resourceId}";
+                }
+                else
+                {
+                    if (reference.Reference.Contains("?", StringComparison.InvariantCulture))
```

Comment:
I'd probably just use StringComparison.Ordinal (byte comparison) unless we have a reason not to